File: src/templates/Dockerfile.ts
FROM node:18-bookworm
WORKDIR /opt/app
COPY package.json ./
ENV NPM_CONFIG_UPDATE_NOTIFIER=false
RUN npm install --no-audit --no-fund
COPY tsconfig.json *.ts ./
RUN npm run build
USER node
CMD [ "npm", "start" ]

File: src/templates/Dockerfile.go
FROM golang:1.14 AS builder
WORKDIR /opt
COPY go.mod ./
RUN go mod download
COPY *.go ./
RUN CGO_ENABLED=0 go build -ldflags="-s -w" -o app

FROM scratch
WORKDIR /opt/app
COPY --from=builder /opt/app .
CMD [ "/opt/app/app" ]
